src/Core/src/Platform/Android/WrapperView.cs

```diff
@@ -83,19 +54,54 @@ public override bool DispatchTouchEvent(MotionEvent e)
 
 		partial void ClipChanged()
 		{
-			_invalidateClip = _invalidateShadow = true;
+			_invalidateClip = true;
 			SetHasClip(Clip is not null);
 		}
 
 		partial void ShadowChanged()
 		{
-			_invalidateShadow = true;
+			if (Shadow?.Paint is { } shadowPaint)
+			{
+				var shadowOpacity = Shadow.Opacity;
+				float radius = Context.ToPixels(Shadow.Radius);
+				float offsetX = Context.ToPixels(Shadow.Offset.X);
+				float offsetY = Context.ToPixels(Shadow.Offset.Y);
```

Comment:
This calls the `Context` property three times in a row. This will call into Java each time, doing the bookkeeping to lookup the C# object.

Can you store `Context` in a local to reuse it?

Reply:
Good catch!